python/ray/data/tests/test_streaming_integration.py

```diff
@@ -478,6 +478,37 @@ def f(x):
         ds2.take_all()
 
 
+def test_e2e_with_unbounded_max_actors(restore_data_context):
+    ray.shutdown()
+    ray.init(num_cpus=10, num_gpus=3)
+    DataContext.get_current().new_execution_backend = True
+    DataContext.get_current().use_streaming_executor = True
+
+    class Actor:
+        def __call__(self, batch):
+            time.sleep(0.1)
+            return batch
+
+    ds = ray.data.range(20, parallelism=20).map_batches(
+        Actor,
+        compute=ray.data.ActorPoolStrategy(min_size=3, max_tasks_in_flight_per_actor=2),
+        num_gpus=1,
+    )
+
+    ds.fully_executed()
+
+    max_tasks_run_for_each_actor = ds._plan.stats().extra_metrics[
```

Comment:
This will be exposed to the user via the stats API. Could we summarize it into a smaller value to avoid stats spam? E.g., max_inflight_tasks.

Reply:
updated to only report this for the test. this is no longer reported as part of the user facing API.